distribution-manifest.md

```diff
@@ -0,0 +1,513 @@
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7"
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 32654,
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 16724,
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 73109,
+      "digest": "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+    }
+  ]
+}
+```
+
+Note that we should get clients to support this generic manifest for images too so we can transition and support upgradeability.
+
+```json
+{
+  "schemaVersion": "3",
+  "mediaType": "application/vnd.oci.distribution.manifest.v1+json",
+  "objects": [
+    {
+      "artifactType": "application/vnd.oci.image",
+      "version": "2",
+      "components": [
+        {
+          "type": "blob",
+          "mediaType": "application/vnd.oci.config",
+          "descriptor": {
+            "mediaType": "application/vnd.oci.image.config.v1+json",
+            "size": 7023,
+            "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7"
+          }
+        },
+        {
+          "type": "blob",
+          "mediaType": "application/vnd.oci.layer",
+          "descriptor": {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+            "size": 32654,
+            "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0"
+          }
+        },
+        {
+          "type": "blob",
+          "mediaType": "application/vnd.oci.layer",
+          "descriptor": {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+            "size": 16724,
+            "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b"
+          }
+        },
+        {
+          "type": "blob",
+          "mediaType": "application/vnd.oci.layer",
+          "descriptor": {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+            "size": 73109,
+            "digest": "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### OCI Image with Alternate Persistence
+
+To support multiple versions of an artifact, the collection of components can be expanded. In the below case, the `"compType": "application/vnd.oci.layer"` is used for standard oci images. With an additional collection of layers persisted as `"compType": "application/vnd.oci.layer.encrypted"`. A client that is unable to parse the encrypted content would pull blobs with: `"compType": "application/vnd.oci.layer"`. While clients capable of pulling encrypted layers pull the encrypted components.
```

Comment:
This doesn't work, you have to use filters for this. This is the current issue, a client that understands both types would have to know out of band that it should only use layers of the "encrypted" type, so it has to know some sort of out of band priority and rules. The format should make it clear that these are strictly alternative, not just a list of items.

Reply:
Two thoughts:
1. Is this a good example of something we can do in phase 2, and we should split the PR into what we'll implement for this summer, vs. later
2. Since there are different `compTypes` or `mediaTypes` in the lastest PR, shouldn't a client know how to process what it knows and ignore the things it doesn't? And, newer versions can choose? I thought the filters were just for architecture, and not for versioning. Or, if a client couldn't know how to process filters, wouldn't they know how to process the `component.mediaTypes`? Or, is there a more generic thing you'd like to accomplish, and we can create a better example?